pandas/core/common.py

```diff
@@ -782,6 +782,13 @@ def take_nd(arr, indexer, axis=0, out=None, fill_value=np.nan,
         will be done.  This short-circuits computation of a mask.  Result is
         undefined if allow_fill == False and -1 is present in indexer.
     """
+
+    if is_categorical(arr):
+        result = take_nd(arr.get_values(), indexer, axis=axis, out=out,
+                         fill_value=fill_value, mask_info=mask_info,
```

Comment:
@jreback I think I could understand. Changed to use ``take_nd``, as ``Categorical`` doesn't have ``take_1d``.